### PR TITLE
Use default port 443 for wss://

### LINKED ISSUE #21: Incorrect default port for wss
The library is using port 80 instead of 443 for wss:// URLs, giving the following errors:

```
2017/06/03 13:20:12 [error] 5776#0: *2 SSL_do_handshake() failed (SSL: error:140770FC:SSL routines:SSL23_GET_
SERVER_HELLO:unknown protocol), context: ngx.timer
```

and

```
2017/06/03 13:10:52 [error] 3441#0: *2 peer closed connection in SSL handshake, context: ngx.timer
ssl handshake failed: handshake failed
```

Using :443 in the URL fixes the problem, like `wss://echo.websocket.org:443`

### PR DESCRIPTION
**Pull request by edubart**
This fixes #21

### CODE REVIEW

File: lib/resty/websocket/client.lua [hunk 1]
```diff
@@ -91,7 +92,11 @@ function _M.connect(self, uri, opts)
     -- ngx.say("port: ", port)
 
     if not port then
-        port = 80
+        if str_sub(uri, 1, 6) == 'wss://' then
```
Comment (agentzh) [inline lib/resty/websocket/client.lua]: The `string.sub()` builtin creates a new Lua string, which is a new GC object, which incurs extra GC overhead and dynamic allocation overhead on this potentially hot code path. Better use the `ngx.re.find()` builtin to do the matching without creating any new GC objects.

**Comment by edubart** [comment]
Changes were made to use re_find instead of string.sub.

**Comment by edubart** [comment]
The testing suit is not working well in my environment (many tests failing) and I cannot fix it because I am not familiar with perl, I won't be able to provide such tests because of this is extra work. Again, I will leave using my fork.

**Comment by agentzh** [comment]
@edubart You don't need to know perl to get this working. Have you seen our tutorial on the testing toolchain here?

https://openresty.gitbooks.io/programming-openresty/content/testing/

Also, you'll find the Travis CI configuration for this project very helpful:

https://github.com/openresty/lua-resty-websocket/blob/master/.travis.yml

This project has been using Travis CI to do regression testing on every commit.

**Comment by agentzh** [comment]
@edubart This is just like you don't need to know C to use the Linux kernel :)

**Pull request closed without merging**